include/dxc/Support/HLSLOptions.h

```diff
@@ -17,10 +17,14 @@
 #include "llvm/ADT/ArrayRef.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Option/ArgList.h"
-#include "dxc/dxcapi.h"
 #include "dxc/Support/HLSLVersion.h"
 #include "dxc/Support/SPIRVOptions.h"
 #include "dxc/Support/DxcOptToggles.h"
+
+#include "dxc/Support/WinIncludes.h"
```

Comment:
There are a few places that add new includes and most common is the addition of this include before dxcapi and friends. Can you explain why that's needed?

Reply:
dxcapi depends on WinIncludes, and we got lucky that it compiled based on the ordering.

I didn't want to modify dxcapi to include WinIncludes directly since dxcapi is part of our public shipped API, and WinIncludes sets macros that alter the windows header definitions in ways that our users might not want.